### PR TITLE
Adding apk module

### PR DESCRIPTION
**Pull request by kbrebanov**
Adding apk module to manage Alpine Linux packages.

Currently supports:
- Installing a package
- Removing a package
- Ensuring a package is at the latest version
- Upgrading all installed packages
- Updating the repository indexes

### CODE REVIEW

**Comment by resmo** [comment]
LGTM! 

One thing I noticed is that you can not pass a list of packages to be installed at once, similar to this `apk add openssh openntp vim` which would reduce the time used for install a bunch of packages.

**Comment by kbrebanov** [comment]
@resmo Added support for installing and removing multiple packages at once.

File: packaging/os/apk.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    module.exit_json(changed=True, msg="removed %s package(s)" % (names))
+        
+# ==========================================
+# Main control flow.
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            state = dict(default='present', choices=['present', 'installed', 'absent', 'removed', 'latest']),
+            name = dict(type='str'),
```
Comment (resmo) [inline packaging/os/apk.py]: use `type='list'` here, so users can pass a list like this:

```
- apk: name={{ item }}
  with_items:
  - foo
  - bar
```

or even comma separated as well (out of the box ansible magic)!

```
- apk: name=foo,bar
```

In your code, you will get python list to use, so your code must be adjusted accordantly.

**Comment by kbrebanov** [comment]
@resmo Added support for 'with_items'

**Comment by resmo** [comment]
+1 from me, handover to core review @gregdek

File: packaging/os/apk.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+- apk: upgrade=yes
+
+# Update repositories as a separate step
+- apk: update_cache=yes
+'''
+
+import os
+import re
+
+APK_PATH="/sbin/apk"
```
Comment (bcoca) [inline packaging/os/apk.py]: use the module's function to find binaries instead of hardcoding this

File: packaging/os/apk.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+def update_package_db(module):
+    cmd = "apk update"
+    rc, stdout, stderr = module.run_command(cmd, check_rc=False)
+    if rc == 0:
+        return True
+    else:
+        module.fail_json(msg="could not update package db")
+
+def query_package(module, name):
+    cmd = "apk -v info --installed %s" % (name)
```
Comment (bcoca) [inline packaging/os/apk.py]: use the commong apk you found already

File: packaging/os/apk.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    rc, stdout, stderr = module.run_command(cmd, check_rc=False)
+    if rc == 0:
+        return True
+    else:
+        return False
+
+def query_latest(module, name):
+    cmd = "apk version %s" % (name)
```
Comment (bcoca) [inline packaging/os/apk.py]: same as above

**Comment by kbrebanov** [comment]
@bcoca Used the module's get_bin_path function

**Pull request merged**